src/lib.rs

```diff
@@ -636,6 +661,13 @@ impl dyn TestStatus {
         run_rustfix(
             &stderr, &stdout, path, comments, revision, config, *mode, extra_args,
         )?;
+
+        // Make sure the `default_file_filter` recognizes that the test has already been run successfully
+        // and avoid running it again.
+        let check_file = config.out_dir.join(path.with_extension("timestamp"));
+        std::fs::create_dir_all(check_file.parent().unwrap()).unwrap();
+        std::fs::File::create(check_file).unwrap();
```

Comment:
> If a test failed last time, then rustc compiletest always reruns it. 

This only happens here, but I realize now that we have an early success path for `run` tests 🙃 so `run` tests are always run again